### ISSUE TITLE
NZBGeek Indexer Uses Truncated Search Query

### ISSUE THREAD

**Issue by ShawnHardwick** [theotherp/nzbhydra2#602]
Running NZBHydra 2.25.0 in docker on a CentOS 7 host.

When executing a search query in the Web UI or external API call, if the indexer is set to NZBGeek, the results returns too many results if there are more than six keywords.

When looking at the debug logs, it appears that the backend Newznab URL is truncating the search query down.

`2020-07-24 18:00:22.577  INFO --- [http-nio-0.0.0.0-5] org.nzbhydra.searching.SearchWeb         : [ID: 53514, Host: 192.168.1.4] New search request: SearchRequest{source=INTERNAL, indexers=[NZBGeek], searchType=SEARCH, category=All, offset=0, limit=100, query=one two three four five six seven eight, identifiers={}}`

`2020-07-24 18:00:22.583  INFO --- [ pool-407-thread-1] org.nzbhydra.indexers.Newznab            : [ID: 53514, Host: 192.168.1.4] NZBGeek: Calling https://api.nzbgeek.info/api?apikey=<apikey>&t=search&extended=1&q=one%20two%20three%20four%20five%20six&password=1&limit=100&offset=0`

This does not appear on any of my other indexers (AnimeTosho, Binsearch, and various Jackett indexers for torrents). You can find example of this on line 19493 timestamp: 2020-07-24 17:58:27.624

[nzbhydra-debuginfos-2020-07-24-14-06.zip](https://github.com/theotherp/nzbhydra2/files/4973642/nzbhydra-debuginfos-2020-07-24-14-06.zip)

**Comment by theotherp** [comment]
Not sure what's your point, sorry. You say more than six words in the query
returns too many results but the search is already being truncated.

**Comment by ShawnHardwick** [comment]
Sure, let me explain with a different example.

If I search for the TV show "Re Zero kara Hajimeru Isekai Seikatsu 2nd Season 02" in the web search of NZBHydra using the NZBGeek indexer, the backend query uses the search criteria `q=Re%20Zero%20kara%20Hajimeru%20Isekai%20Seikatsu` which has truncated any words past the first 6. In this case it removed `2nd Season 02`. Since it removed this additional context, the end user sees too many results. The overall issue is that what the user/api asks to search is being truncated for the NZBGeek indexer.

**Comment by albino1** [comment]
That's intentional because NZBGeek's API is borked if you use more than 6 keywords, so @theotherp added that custom code ages ago to deal with their broken API.  In fact, it was probably way back in NZBHydra 1.

I just checked and it's still broken with more than 6 keywords.  You can test this yourself fairly easily:

Works 6 keywords: `https://api.nzbgeek.info/api?apikey=<YOURAPIKEY>&t=search&extended=1&q=Re%20Zero%20kara%20Hajimeru%20Isekai%20Seikatsu&password=1&limit=100&offset=0`

Fails 9 keywords: `https://api.nzbgeek.info/api?apikey=<YOURAPIKEY>&t=search&extended=1&q=Re%20Zero%20kara%20Hajimeru%20Isekai%20Seikatsu%202nd%20Season%2002&password=1&limit=100&offset=0`

**Comment by ShawnHardwick** [comment]
Oh wow, you are right. Dang that sucks. Time to find a new indexer. My Sonarr instance crawls to a halt because its processing 1500 releases from the truncation.

**Comment by ShawnHardwick** [comment]
I submitted a bug report to NZBGeek and they have been aware of the issue, but hopefully my report might place the issue a little higher. For anyone with NZBGeek VIP forum access, the thread is here: https://nzbgeek.info/showthread.php?tid=13643&pid=48846#pid48846

> > We will look at handling this our end on the API side as a workaround and when I have a working solution we will ask NZBHydra to remove their truncate fix.

As far as my Sonarr instance crawling to a halt, I realized that the Sonarr web UI stops querying for results after 3 minutes when using the manual search method. Side effect having to process too many releases.

I'll leave it up to you if wish to keep this issue open or not.
